--- a/drivers/gpu/drm/panel/panel-simple.c	2020-02-11 19:35:55.000000000 +0700
+++ b/drivers/gpu/drm/panel/panel-simple.c	2020-02-14 22:47:48.882881488 +0700
@@ -2271,6 +2271,29 @@
 	.bus_format = MEDIA_BUS_FMT_RGB888_1X24,
 };
 
+static const struct drm_display_mode olimex_lcd_olinuxino_7_mode = {
+	.clock = 25500,
+	.hdisplay = 800,
+	.hsync_start = 800 + 10,
+	.hsync_end = 800 + 10 + 30,
+	.htotal = 800 + 10 + 30 + 10,
+	.vdisplay = 480,
+	.vsync_start = 480 + 10,
+	.vsync_end = 480 + 10 + 5,
+	.vtotal = 480 + 10 + 5 + 5,
+	.vrefresh = 60,
+};
+
+static const struct panel_desc olimex_lcd_olinuxino_7 = {
+	.modes = &olimex_lcd_olinuxino_7_mode,
+	.num_modes = 1,
+	.size = {
+		.width = 159,
+		.height = 95,
+	},
+	.bus_format = MEDIA_BUS_FMT_RGB888_1X24,
+};
+
 /*
  * 800x480 CVT. The panel appears to be quite accepting, at least as far as
  * pixel clocks, but this is the timing that was being used in the Adafruit
@@ -3292,6 +3315,9 @@
 		.compatible = "olimex,lcd-olinuxino-43-ts",
 		.data = &olimex_lcd_olinuxino_43ts,
 	}, {
+		.compatible = "olimex,lcd-olinuxino-7",
+		.data = &olimex_lcd_olinuxino_7,
+	}, {
 		.compatible = "ontat,yx700wv03",
 		.data = &ontat_yx700wv03,
 	}, {
